### PR TITLE
Read the room version from database when fetching events

### PR DESCRIPTION
**Pull request by richvdh**
This is a precursor to giving EventBase objects the knowledge of which room version they belong to.

### CODE REVIEW

**Comment by richvdh** [comment]
there is a lot of this sort of thing going on:

```
2020-02-11 12:11:20,894 - synapse.app.synchrotron - 434 - ERROR - replication-RDATA-events-54 - Error processing replication
...
synapse.api.errors.NotFoundError: 404: Could not find event $EHbO6nL4Orti5xt2HNHOB2HB9YH3v6AgByrv5EDBRPY
```

I guess we aren't always making sure that the room is written to the db before the event lands.

**Comment by richvdh** [comment]
this is ready for review now I think!

File: synapse/storage/data_stores/main/events_worker.py [hunk 1]
```diff
@@ -580,8 +583,49 @@ def _get_events_from_db(self, event_ids, allow_rejected=False):
...
+                    )
+                    continue
+
+                # take a wild stab at the room version based on the event format
+                if format_version == EventFormatVersions.V1:
+                    room_version = RoomVersions.V1
+                elif format_version == EventFormatVersions.V2:
+                    room_version = RoomVersions.V3
+                else:
+                    room_version = RoomVersions.V5
```
Comment (erikjohnston) [inline synapse/storage/data_stores/main/events_worker.py]: Is the idea that for out of band memberships after this commit will have a room version stored? I.e. we should never get here once for later room versions?

Comment (richvdh) [reply synapse/storage/data_stores/main/events_worker.py]: well, out-of-band-memberships since #6983. But yes.

**Pull request merged**